src/CLIService/date.ts

```diff
@@ -69,23 +70,34 @@ const timePeriods = [
   },
 ];
 
-export const promtTimePeriod = async (field: string): Promise<number> => {
-  // because of https://github.com/enquirer/enquirer/issues/405
-  // @ts-expect-error
-  const { period } = await prompt<{ period: number }>({
-    type: "select",
-    name: "period",
-    message: `Please enter ${field} unit`,
-    choices: timePeriods,
-  });
+export const promptTimePeriod = async (field: string, period?: number, periodValue?: number): Promise<number> => {
+  if (period === undefined) {
+    // because of https://github.com/enquirer/enquirer/issues/405
+    // @ts-expect-error
+    period = (await prompt<{ period: number }>({
```

Comment:
the comment isn't specific to this line:
too much of refactoring now, but I'm a bit confused by duplications of these params here and at `/src/config.ts`. 
Could we somehow reuse same config for these configuration properties in the future?

Reply:
I see. This is reasonable. 

However, this approach may be fragile in some sense because we may just lose some validations or some parameter input since we need to manually be sure it exists. Validations exist in several places that may lead to inconsistencies (for instance, if you specify a parameter via arguments -> <behavior 1>, but if you specify it later -> <behavior 2>).
Again, it is more like improvement/nice-to-have suggestion.

Reply:
btw this logic already exists.
https://github.com/streamflow-finance/bulk-streams-cli/blob/main/src/CLIService/index.ts#L52,
but it is executed only on startup. 
CLI can just launch it with small adjustments when we need it again.